spec/lib/avalon/build_utils_spec.rb

```diff
@@ -0,0 +1,166 @@
+    contents_arr.push "blahblah"
+  end
+
+# TODO: ADD TESTS FOR READING CONFIG FILE
+
+  describe '#detect_version' do
+    it 'Detects 3 part version correctly with contents array' do
+      version = "1.2.3"
+      contents_arr = get_contents_array(version)
+
+      expect(Utils.detect_version(contents_arr)).to eq('1.2.3')
+
+    end
+
+    it 'Detects 3 part version correctly with contents string' do
+      version = "1.2.3"
+      contents_str = get_contents_string(version)
+      #puts "#{contents_str}"
+      expect(Utils.detect_version(contents_str)).to eq('1.2.3')
+    end
+
+    it 'Detects 4 part version correctly' do
+      version = "1.2.3.4"
+      contents_arr = get_contents_array(version)
+
+      expect(Utils.detect_version(contents_arr)).to eq('1.2.3.4')
+
+    end
+
+    it 'Detects >4 part version correctly (not supported)' do
+      version = "1.2.3.4.5"
+      contents_arr = get_contents_array(version)
+
+      expect(Utils.detect_version(contents_arr)).to eq('')
+
+    end
+
+    it 'Detects 2 part version correctly (not supported)' do
+      version = "1.2"
+      contents_arr = get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('')
+    end
+
+    it 'Detects 1 part version correctly (not supported)' do
+      version = "1"
+      contents_arr = get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('')
+    end
+
+    it 'Ignores commented lines in config file contents' do
+      version = "3.14.1"
+      contents_arr = []
+      contents_arr.push "# VERSION = '2.0.1'"
+      contents_arr.concat get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('3.14.1')
+    end
+
+    describe '#get_tags' do
+      it 'Returns an empty array if no branch nor top_level options' do
+        version = "1.2.3"
+        expect(Utils.get_tags(version)).to eq([])
+      end
+      it 'Outputs correct tags for a 3-part version and top_level true' do
+        version = "1.2.3"
+        options = {"top_level": true}
+        expect(Utils.get_tags(version, options)).to eq(["1.2.3"])
+      end
+
+      it 'SPLITS and outputs correct tags for a 3-part version and no additional tags' do
+        version = "1.2.3"
+        options = {"top_level": true, "split": true}
+        expect(Utils.get_tags(version, options)).to eq(["1", "1.2", "1.2.3"])
+      end
+
+      it 'Outputs correct tags for a 3-part version and a branch tag' do
+        version = "3.2.1"
+        options = {"split": false, "branch": "staging"}
+        expect(Utils.get_tags(version, options)).to eq(["3.2.1-staging", "staging"])
+      end
+
+      it 'SPLITS and Outputs correct tags for a 3-part version and a branch tag TOP LEVEL production branch' do
+        version = "3.2.1"
+        options = {"split": true, "branch": "production", "top_level": true}
+        expect(Utils.get_tags(version, options)).to eq(["3-production", "3.2-production", "3.2.1-production", "production"])
```

Comment:
Given that `top_level` returns the undecorated tags when there isn't a branch, I would expect this to return the union of the branch non-top level and non-branched top level tags.  Does that make sense?  Or does the `branch` option override the `top_level` option and only one of them can be used at a time?

```suggestion
        expect(Utils.get_tags(version, options)).to eq(["3", "3.2", "3.2.1", "3-production", "3.2-production", "3.2.1-production", "production"])
```

Reply:
Good eye, fixing this in a new commit